app/models/customer_account.rb

```diff
@@ -65,6 +65,14 @@ def self.find_by_reference(reference)
       nil
     end
 
+    # Find customer account by password reset token provided in email's link
+    # Used in reset password scenario
+    def self.find_by_token(token)
```

Comment:
Will create separate PR for that since need to refactor also existing `find_by_email` and `find_by_reference`